### ISSUE TITLE
ttnn.permute fails with UnboundLocalError

### ISSUE THREAD

**Issue by Sudharsan-V** [tenstorrent-metal/tt-metal#5150]
**Describe the bug**

In the `ttnn/ttnn/operations/data_movement.py` file, specifically in the method `def permute`, the input tensor is received as `input_tensor`. However, within the function, it is referred to as a tensor, causing an UnboundLocalError.

**Screenshots**
```
            def torch_permute(tensor, order):
                return tensor.permute(order).contiguous().clone()
    
>           tensor = ttnn.to_torch(tensor)
E           UnboundLocalError: local variable 'tensor' referenced before assignment

ttnn/ttnn/operations/data_movement.py:158: UnboundLocalError
```

**Additional context**
Replacing `tensor = ttnn.to_torch(tensor)` to `tensor = ttnn.to_torch(input_tensor)` resolves this issue.

**Comment by arakhmati** [comment]
Is there a PR with the fix? If yes, we should merge it in

**Comment by saichandax** [comment]
@arakhmati , we don't have a PR for the fix.
If you would like us to create one, we'll.

**Comment by Sudharsan-V** [comment]
@arakhmati, fix for this issue is added in the PR [#5175](https://github.com/tenstorrent-metal/tt-metal/pull/5175)

**Comment by tarafdarTT** [comment]
since this has been addressed I'm closing this.
